configure.ac

```diff
@@ -1890,13 +1909,13 @@ then
 fi
 
 # WASM flags
-# TODO: Add -s MAIN_MODULE=2 for dlopen() support.
-# The option disables code elimination, which increases code size of main
-# binary. All objects must be built with -fPIC.
 AS_CASE([$ac_sys_system/$ac_sys_emscripten_target],
   [Emscripten/browser*], [
     LDFLAGS_NODIST="$LDFLAGS_NODIST -s ALLOW_MEMORY_GROWTH=1"
     LINKFORSHARED="--preload-file \$(WASM_ASSETS_DIR)"
+    AS_VAR_IF([enable_wasm_dynamic_linking], [yes], [
+      AS_VAR_APPEND([LINKFORSHARED], [" -s MAIN_MODULE=1"])
```

Comment:
What's the reason for not using `-s MAIN_MODULE=2` to reduce code size? Would that reason be worth a comment somewhere?

Reply:
Using `-s MAIN_MODULE=2` requires you to build the core Python interpreter after building all the C extensions and then unpack all the wheels into where you plan to install them in the Emscripten filesystem, then list out all the different `.so` files in any of the packages as arguments to the linker. This is a bit impractical (scipy by itself has 99 `.so` files). `-s MAIN_MODULE=2` is really intended for code splitting not for shared libraries used with `dlopen`.

Reply:
Thanks for the insight!